core/src/services/onedrive/builder.rs

```diff
@@ -20,25 +20,46 @@ use std::fmt::Debug;
 use std::fmt::Formatter;
 
 use log::debug;
+use serde::Deserialize;
 
 use super::backend::OnedriveBackend;
 use crate::raw::normalize_root;
 use crate::raw::HttpClient;
 use crate::Scheme;
 use crate::*;
 
+/// Config for [OneDrive](https://onedrive.com) backend support.
+#[derive(Default, Deserialize)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct OnedriveConfig {
```

Comment:
Please also refactor `Builder::from_map` for `OnedriveBuilder`.

Reply:
It should be fixed